File: typescript/tests/palindromeCheck.test.ts
import { describe, expect, it } from "vitest";

import { palindromeCheck } from "../src/strings/palindromeCheck.js";

describe("palindromeCheck", () => {
  it("simple", () => {
    expect(palindromeCheck("racecar")).toBe(true);
  });

  it("with punctuation", () => {
    expect(palindromeCheck("A man, a plan, a canal: Panama")).toBe(true);
  });

  it("not palindrome", () => {
    expect(palindromeCheck("robot")).toBe(false);
  });

  it("empty", () => {
    expect(palindromeCheck("")).toBe(true);
  });

  it("mixed case", () => {
    expect(palindromeCheck("Aa")).toBe(true);
  });
});

File: java/src/test/java/com/example/livecoding/strings/PalindromeCheckTest.java
package com.example.livecoding.strings;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class PalindromeCheckTest {

    @Test
    void simple() {
        assertTrue(PalindromeCheck.palindromeCheck("racecar"));
    }

    @Test
    void withSpacesAndPunctuation() {
        assertTrue(PalindromeCheck.palindromeCheck("A man, a plan, a canal: Panama"));
    }

    @Test
    void notPalindrome() {
        assertFalse(PalindromeCheck.palindromeCheck("robot"));
    }

    @Test
    void empty() {
        assertTrue(PalindromeCheck.palindromeCheck(""));
    }

    @Test
    void mixedCaseIgnored() {
        assertTrue(PalindromeCheck.palindromeCheck("Aa"));
    }
}
